cmd/heimdalld/cmd/migration/verify/verify.go

```diff
@@ -873,3 +866,83 @@ type validatorBasicInfo struct {
 	nonce  uint64
 	jailed bool
 }
+
+// verifyMilestones verifies the milestones in the genesis files by comparing the data in both versions
+func verifyMilestones(ctx types.Context, app *heimdallApp.HeimdallApp, hv1Genesis map[string]interface{}) error {
+	checkpointModule, ok := hv1Genesis["app_state"].(map[string]interface{})["checkpoint"]
+	if !ok {
+		return fmt.Errorf("checkpoint module not found in app_state")
+	}
+
+	checkpointData, ok := checkpointModule.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("failed to cast checkpoint module data")
+	}
+
+	milestones, ok := checkpointData["milestones"].([]interface{})
+	if !ok {
+		return fmt.Errorf("milestones not found or invalid format")
+	}
+
+	if milestones == nil {
+		milestones = []interface{}{}
```

Comment:
I think this is not enough for the following case (which I had to fix for `txSequences` in `topup` module.
If we don't have milestones from v1, I would handle both cases in the json file, meaning `milestones = null` and `milestones = []`.

Reply:
(this can be verified by running the `TestRunMigrationVerification` with modified data sets for v1 milestones.

Reply:
Fixed.